### PR TITLE
Fixes the weather demo build

### PR DESCRIPTION
**Pull request by andreban**
- Adds the declare global clause to `types.d.ts` to make the defined types available.
 - Changes from `AISession` to `AIAssistant` in `BuiltinPrompting` to match the new API.
 - Changes the return type from `streamingPrompt` to match the new API.

### CODE REVIEW

**Comment by tomayac** [comment]
Can you instead go into https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70426 and copy over from there? If there're issues, please review them directly over there.

**Comment by andreban** [comment]
> Can you instead go into [DefinitelyTyped/DefinitelyTyped#70426](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70426) and copy over from there? If there're issues, please review them directly over there.

I'd rather us merging this now and have working code in the repo and move to DefinitelyTyped as dependency once the definitions are published.

**Comment by tomayac** [comment]
Oh, I wasn't saying we not merge this. Just, if you make a change, be sure to make it on the other repo.

**Review by tomayac — APPROVED** [review]
LGTM

**Comment by andreban** [comment]
> Oh, I wasn't saying we not merge this. Just, if you make a change, be sure to make it on the other repo, too.

Ah, got it. I'll see if there's any way I can help there, but I'm no expert in TS types and how they work once landed on DefinitelyTyped .

**Pull request merged**
